src/automove.rs

```diff
@@ -60,40 +61,63 @@ impl AutoMove {
     /// This doesn't actually move the files but each entry contains the
     /// current file path and the new wanted file path.
     pub fn run(&self) -> Vec<AutoMoveResult> {
-        let mut results = Vec::new();
-        for rule in self.rules.iter() {
-            if let Some(parent) = &self.parent {
-                if !rule.directory.starts_with(parent) {
-                    continue;
+        self.rules
+            .iter()
+            .filter(|rule| {
+                if let Some(parent) = &self.parent {
+                    rule.directory.starts_with(parent)
+                } else {
+                    true
                 }
-            }
-            results.push(rule.run());
-        }
-        results
+            })
+            .map(AutoMoveRule::run)
+            .collect()
     }
 
     /// Checks if any file would be moved if this were to be run
     pub fn would_move_any(&self) -> bool {
-        for rule in self.rules.iter() {
-            if rule.would_move() {
-                return true;
-            }
-        }
-        false
+        self.rules.iter().any(AutoMoveRule::would_move)
     }
 
     /// Counts how many files would be moved across rules if this were to be run
     pub fn count_move(&self) -> usize {
         self.rules.iter().fold(0, |a, b| a + b.count_move())
     }
+
+    pub fn check_empty(&self, config: &Config) -> anyhow::Result<()> {
+        if self.rules.is_empty() {
+            let w = if config.settings.color {
+                "(!)".bold()
+            } else {
+                "(!)".into()
+            };
+            bail!("{} No auto-move rules were configured.", w)
+        }
+        Ok(())
+    }
```

Comment:
This now exits the program with error code 1 and with the message "Error: (!) No auto-move rules were configured.". I think the prefix `Error: ` is redundant, since there is already `(!)`, and this message is more of a warning rather than an error. If you think that's intentional then sure, it can stay that way- just making sure it wasn't overlooked